CMakeLists.txt

```diff
@@ -58,26 +59,40 @@ include_directories(SYSTEM "lib/easyloggingpp/src" "lib/better-enums/")
 if (COMPILE_TESTS)
     add_subdirectory("lib/googletest")
 endif()
-if (COMPILE_PYBIND)
-    add_subdirectory("lib/pybind11")
-endif()
 
 set( CMAKE_BUILD_TYPE_COPY "${CMAKE_BUILD_TYPE}" )
 set( CMAKE_BUILD_TYPE "Release" )
 option(build_static_lib "Build easyloggingpp as a static library" ON)
-add_subdirectory("lib/easyloggingpp")
+if (BUILD_PIP_PACKAGE)
+    # EXCLUDE_FROM_ALL is undocumented and used to solve issues with easyloggingpp access rights
+    add_subdirectory("lib/easyloggingpp" EXCLUDE_FROM_ALL)
+else ()
+    add_subdirectory("lib/easyloggingpp")
+endif ()
 set( CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE_COPY} )
 
 add_subdirectory("src/core")
 
 if (COMPILE_TESTS)
     add_subdirectory("src/tests")
 endif()
+
 if (UNPACK_DATASETS)
     add_subdirectory("datasets")
 endif()
+
 add_subdirectory("cfg")
 
-if (COMPILE_PYBIND)
-    add_subdirectory("src/python_bindings")
+
+SET(BINDINGS_NAME desbordante)
+
+if (BUILD_PIP_PACKAGE OR COMPILE_BINDINGS)
+    add_subdirectory("lib/pybind11")
+    file(GLOB_RECURSE sources "src/python_bindings/*.h*" "src/python_bindings/*.cpp*")
+    pybind11_add_module(${BINDINGS_NAME} ${sources})
+    target_link_libraries(${BINDINGS_NAME} PRIVATE ${CMAKE_PROJECT_NAME}_lib)
+endif ()
+
+if (BUILD_PIP_PACKAGE)
+    install(TARGETS ${BINDINGS_NAME} LIBRARY DESTINATION .)
```

Comment:
Perhaps it would make sense to have an `ENABLE_PYTHON` option in the main CMakeLists file, and then an `INSTALL_PYTHON` option in `src/python_bindings/CMakeLists.txt`?

Reply:
There is about zero good documentation for pybind11 + scikit-build-core, so I have no idea how to do this without pain

Reply:
I did this for you.